### PR TITLE
Implement Request

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
I'm just hacking around, might benefit from #148.
Also this approach passes an encoded JSON (which requires unmarshaling on the Go side).
Now it's possible to do stuff like:
```typescript
async function tests_fetch() {
    let req = new Request("http://httpbin.org/headers", {method: "GET"});
    const response = await fetch(req);
    const json = await response.json();
    console.log(json)
}
```
Feedback/ideas appreciated 👍

### CODE REVIEW

File: runtime.ts [hunk 1]
```diff
@@ -195,7 +195,8 @@ class Compiler {
     inlineSourceMap: true,
     lib: ["es2017"],
     inlineSources: true,
-    target: ts.ScriptTarget.ES2017
+    target: ts.ScriptTarget.ES2017,
+    esModuleInterop: true
```
Comment (qti3e) [inline runtime.ts]: Why is this needed?

Comment (matiasinsaurralde) [reply runtime.ts]: Oh left that from a previous experiment, not required

File: fetch.ts [hunk 2]
```diff
@@ -90,12 +90,36 @@ class FetchResponse implements Response {
...
+      opts = {};
+    }
+    this.opts = opts;
+    this.opts.url = url;
+    // Object.assign(this.opts, opts);
+  }
+}
+
 let nextFetchId = 0;
 //TODO implements Request
```
Comment (jedahan) [inline fetch.ts]: Can this line be removed now?

Comment (matiasinsaurralde) [reply fetch.ts]: Will remove it!

**Comment by ry** [comment]
@matiasinsaurralde as you know, rust rewrite. I can't land this now. Closing this PR. Hopefully we'll get back to fixing up this stuff in the next week or two.
